src/odemis/acq/test/acq_test.py

```diff
@@ -486,5 +484,306 @@ def on_progress_update(self, future, start, end):
+        cls.light = model.getComponent(role="light")
+        cls.light_filter = model.getComponent(role="filter")
+        cls.ebeam = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.SEM_focuser = model.getComponent(role="ebeam-focus")
+        cls.FM_focuser = model.getComponent(role="focus")
+
+        cls.FM_focus_pos = 0.5e-6  # arbitrary current focus position
+        cls.SEM_focus_pos = 0.1  # arbitrary current position
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        del cls.light
+        test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+        self._nb_updates = 0
+        self.streams = []
+
+        self.FM_focuser.moveAbs({"z": self.FM_focus_pos}).result()
+        self.SEM_focuser.moveAbs({"z": self.SEM_focus_pos}).result()
+
+    @unittest.skip("Only zstack")
+    def test_only_FM_streams_without_zstack(self):
+        # create streams
+        s1 = stream.FluoStream(
+            "fluo1", self.ccd, self.ccd.data, self.light, self.light_filter
+        )
+        s1._focuser = self.FM_focuser
+        s1.excitation.value = sorted(s1.excitation.choices)[0]
+        s2 = stream.FluoStream(
+            "fluo2", self.ccd, self.ccd.data, self.light, self.light_filter
+        )
+        s2._focuser = self.FM_focuser
+        s2.excitation.value = sorted(s2.excitation.choices)[-1]
+        self.streams = [s1, s2]
+
+        # mock the zlevels dictionary
+        zlevels = {}
+        for s in self.streams:
+            zlevels[s] = [s.focuser.position.value["z"]]
+
+        # start the acquisition
+        f = acqmng.acquireZStack(self.streams, zlevels)
+        f.add_update_callback(self._on_progress_update)
+
+        # get the data
+        data, exp = f.result()
+        self.assertIsNone(exp)
+
+        for d in data:
+            self.assertIsInstance(d, model.DataArray)
+            # since no zstack, the center has 2 components
+            self.assertEqual(len(d.metadata[model.MD_POS]), 2)
+            # since no zstack, the pixel size has 2 components
+            self.assertEqual(len(d.metadata[model.MD_PIXEL_SIZE]), 2)
+
+        # 2 streams, then 2 acquisitions
+        self.assertEqual(len(data), 2)
+
+        # 2 streams, 1 updates per stream, so 2 updates
+        self.assertGreaterEqual(self._nb_updates, 2)
+
+    def _on_progress_update(self, f, s, e):
+        self._nb_updates += 1
+
+    @unittest.skip("Only zstack")
+    def test_only_SEM_streams_without_zstack(self):
+        # create streams
+        sems = stream.SEMStream("sem", self.sed, self.sed.data, self.ebeam)
+        sems._focuser = self.SEM_focuser
+        self.streams = [sems]
+
+        # mock the zlevels dictionary
+        zlevels = {}
+        for s in self.streams:
+            zlevels[s] = [s.focuser.position.value["z"]]
+
```

Comment:
Please test `estimateZStackAcquisitionTime()` too, in the different test cases. You can just check that it returns a value above an arbitrary value (eg, > 2s).